os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/serial_api.c

```diff
@@ -741,7 +740,6 @@ void serial_putc(serial_t *obj, int c)
 {
 	PMBED_UART_ADAPTER puart_adapter = &(uart_adapter[obj->uart_idx]);
 
-	while (!serial_writable(obj));
```

Comment:
Serial driver checks `uart_txready`. It looks different with this `serial_writable`. 
The `uart_txready` returns sw status, `serial_writable` does hw status.
Do you think sw status can guarantee serial status instead of hw status?

UART issue will be critical. Please let me know your test way and result.

Reply:
I have made this changes with reference to tp1x where it fixes uart data loss during flash operation.

https://github.com/Samsung/TizenRT/commit/8ee1b95bb635454c31dbbc646eb747a6c46d585d#diff-ce9702b589ceb0508b8098f5ce272259dd14285ee17095d7e34a3aa6bd05779f

I will perform some stress test and share you the results.

Reply:
Hi @sunghan-chang,

I have verified this PR with a loopback test sending 160bytes of data in a while loop. There is no data loss in a long run test.
The UART TX FIFO has a depth of 16 bytes. In the TX interrupt handler (TxIrq), data is transmitted in 16-byte chunks, matching the FIFO size. This ensures optimal use of the FIFO and minimizes the number of interrupts required.

The TX interrupt is only triggered when the TX FIFO is completely empty. This guarantees that when uart_xmitchars() is called in response to the interrupt, the FIFO has enough space to accept up to 16 new bytes of data.

For RX case, rtl8730e_up_rxavailable already ensures that there is available data in fifo before calling receive.

Reply:
Update:
currently running longrun with below setting
UART loopback test + smartfs powercut running concurrently.
No data loss so far.

Reply:
@zhongnuo-tang Could you update the longrun test status?